openverse_catalog/dags/providers/provider_api_scripts/finnish_museums.py

```diff
@@ -1,167 +1,128 @@
-    query_params = default_params.copy()
-    query_params.update(
-        {
-            "page": page,
-            "filter[]": [f'format:"{FORMAT_TYPE}"', f'building:"{building}"'],
-        }
-    )
-
-    return query_params
-
-
-def _get_object_list_from_json(json_resp):
-    if (
-        json_resp is None
-        or str(json_resp.get("status")).lower() != "ok"
-        or json_resp.get("records") is None
-        or len(json_resp.get("records")) == 0
-    ):
-        object_list = None
-    else:
-        object_list = json_resp.get("records")
-
-    return object_list
-
-
-def _process_object_list(object_list):
-    total_images = 0
-    if object_list is not None:
-        for obj in object_list:
-            total_images = _process_object(obj)
-
-    return total_images
-
-
-def _process_object(obj, sub_providers=SUB_PROVIDERS, provider=PROVIDER):
-    total_images = 0
-    license_url = get_license_url(obj)
-    if license_url is None:
-        return None
-    foreign_identifier = obj.get("id")
-    title = obj.get("title")
-    building = obj.get("buildings")[0].get("value")
-    source = next((s for s in sub_providers if building in sub_providers[s]), provider)
-    foreign_landing_url = _get_landing(obj)
-    raw_tags = _get_raw_tags(obj)
-    image_list = obj.get("images")
-    for img in image_list:
-        image_url = _get_image_url(img)
-        total_images = image_store.add_item(
-            license_info=get_license_info(license_url=license_url),
-            foreign_identifier=foreign_identifier,
-            foreign_landing_url=foreign_landing_url,
-            image_url=image_url,
-            title=title,
-            source=source,
-            raw_tags=raw_tags,
+class FinnishMuseumsDataIngester(ProviderDataIngester):
+    providers = {"image": PROVIDER}
+    endpoint = "https://api.finna.fi/api/v1/search"
+    batch_limit = 100
+    delay = 5
+    format_type = "0/Image/"
+
+    def ingest_records(self, **kwargs):
+        for building in BUILDINGS:
+            logger.info(f"Obtaining images of building {building}")
+            super().ingest_records(building=building)
+
+    def get_next_query_params(self, prev_query_params, **kwargs):
+        if not prev_query_params:
+            building = kwargs.get("building")
+            return {
+                "filter[]": [f'format:"{self.format_type}"', f'building:"{building}"'],
+                "limit": self.batch_limit,
+                "page": 1,
+            }
+        return {**prev_query_params, "page": prev_query_params["page"] + 1}
+
+    def get_media_type(self, record):
+        return "image"
+
+    def get_batch_data(self, response_json):
+        if (
+            response_json is None
+            or str(response_json.get("status")).lower() != "ok"
+            or response_json.get("records") is None
+            or len(response_json.get("records")) == 0
+        ):
+            return None
+
+        return response_json.get("records")
```

Comment:
```suggestion
        return response_json["records"]
```
Tiny nit: we have already tested that `response_json` has a key of `records`, and the value's length is not 0.

Reply:
True!